Fix broken Jetpack e2e tests

rcanepa opened the pull request:
#### Changes proposed in this Pull Request

* After the launch of Offer Reset, a considerable number of e2e tests started to fail. This is because this project introduced a new Plans grid and a new pricing page, and these changes weren't reflected in our tests suite.
* This PR is an attempt to fix all broken tests and make them work with the new UIs.

#### Implementation notes

* I would very appreciate feedback regarding the usage of classes and data attributes to select elements in the DOM. When we implemented the new Jetpack Plans page, we didn't add any data attributes to support e2e tests. Is that something we should care about? Since we don't have those, I had to rely on text content (which I think is a good idea) and CSS classes (which I don't think is a good idea). What are your thoughts on this matter?
* Many tests were almost the same between Jetpack and non-Jetpack sites since they shared the same plans grid and almost the same plan names. This has changed dramatically. The plans grid is no longer the same and neither the Jetpack plans. Because of this, the fixes I introduced here look like patches instead of long term solutions. I would say that this is okay for now because we are still in the process of changing the Jetpack Plans page. Once we reach an stable version of the page, I believe we should take a look at these tests again and introduce better abstractions if needed.

#### Testing instructions

* Code review.
* Do tests pass in CircleCI?

Fixes 1164141197617539-as-1198203233393040

rcanepa commented on test/e2e/lib/pages/external/jetpackcom-pricing-page.js:
```diff
@@ -12,13 +12,14 @@ import * as driverHelper from '../../driver-helper';
 	constructor( driver, url ) {
 		if ( ! url ) {
-			url = 'https://jetpack.com/pricing/';
+			url = 'https://cloud.jetpack.com/pricing/';
```
This is the new location of the Pricing page.

rcanepa commented on test/e2e/specs-jetpack-calypso/wp-jetpack-plans-spec.js:
```diff
@@ -68,13 +70,13 @@ describe( `[${ host }] Jetpack Plans: (${ screenSize }) @jetpack`, function () {
 			return await jetpackDashboard.clickUpgradeNudge();
 		} );
 
-		step( 'Can click upgrade on Jetpack landing page', async function () {
-			const searchLandingPage = await JetpackComSearchLandingPage.Expect( driver );
-			return await searchLandingPage.upgrade();
-		} );
-
-		step( 'Can then see secure payment component', async function () {
-			return await SecurePaymentComponent.Expect( driver );
```
This upgrade link no longer redirects users to Jetpack Search Landing page, instead, it redirects users directly to the checkout page with Jetpack Search in the cart.

rcanepa commented on test/e2e/lib/components/secure-payment-component.js:
```diff
@@ -26,7 +27,8 @@ export default class SecurePaymentComponent extends AsyncBaseContainer {
 		this.premiumPlanSlug = getJetpackHost() === 'WPCOM' ? 'value_bundle' : 'jetpack_premium';
-		this.businessPlanSlug = getJetpackHost() === 'WPCOM' ? 'business-bundle' : 'jetpack_business';
+		this.businessPlanSlug =
+			getJetpackHost() === 'WPCOM' ? 'business-bundle' : 'jetpack_security_daily';
```
Jetpack Professional (business) is no longer a thing so I replaced its slug with one of an actual Jetpack plan.

rcanepa commented on test/e2e/lib/components/secure-payment-component.js:
```diff
@@ -76,6 +78,19 @@ export default class SecurePaymentComponent extends AsyncBaseContainer {
 		// SecurePaymentComponent.completeTaxDetailsInContactSection.
 		await this.completeTaxDetailsForCreditCard( { cardPostCode, cardCountryCode } );
 
+		const creditCardHandleSelector = By.css( 'label[for="card"]' );
+		if ( currentScreenSize() === 'mobile' ) {
+			await driverHelper.scrollIntoView( this.driver, creditCardHandleSelector );
+		}
+
+		// Sometimes the credit card form will be closed and it will require a click to be opened.
+		// This can happen when users have a credit card already associated with their account.
+		await driverHelper.selectElementByText(
+			this.driver,
+			creditCardHandleSelector,
+			'Credit or debit card'
+		);
+
```
I noticed that a test didn't pass because it got stuck on the entering credit card details step. It happened because the testing user already had some credit cards stored so the credit card form was closed/hidden. To complete the test, I had to make the client to search for the credit card row, click it (to open the form), and then make it fill in the data.

brbrr commented on test/e2e/lib/pages/plans-page.js:
```diff
@@ -64,15 +66,23 @@ export default class PlansPage extends AsyncBaseContainer {
 		// Wait a little for loading animation
 		await this.driver.sleep( 1000 );
+
+		// Jetpack Business (Professional) no longer exist.
```
Let's remove this, since this comment does not help much. 

```suggestion
```

brbrr commented on test/e2e/lib/pages/external/jetpackcom-pricing-page.js:
```diff
@@ -12,13 +12,14 @@ import * as driverHelper from '../../driver-helper';
-		return await driverHelper.clickWhenClickable( this.driver, buyPremiumSelector );
+	// To buy a Jetpack Plan, we search for `planName` in every CTA on the page.
+	async buyJetpackPlan( planName ) {
+		const planCTA = By.xpath( `//button[@type='button' and contains(., 'Get ${ planName }')]` );
```
Generally, test locators are not recommended, since the text could change more often, which could lead to test breakage more often. Also, they might not work in a non-english locale.

Can we, instead of using text locators, add data attributes to the components we are interested in?

I see similar text locators in few other places, so it might make sense to add data attributes there as well.

rcanepa replied:
Since this page will probably be changed a lot in the next coming weeks, I think it will be better if we use data-attributes. They will also serve as warning for developers that they might break tests if they don't preserve them.

brbrr reviewed (changes requested):
Changes looks good!

Although, I think we should replace text locators with css locators, as explained in comment below.

rcanepa commented on test/e2e/lib/components/secure-payment-component.js:
```diff
@@ -2,11 +2,12 @@
  * Internal dependencies
  */
+// eslint-disable-next-line wpcalypso/no-package-relative-imports
+import config from 'config';
```
Great question. I changed it because I'm used to see that import in the internal dependencies block ([here is an example](https://github.com/Automattic/wp-calypso/blob/master/client/login/controller.js#L11)) and because eslint was complaining about it showing me a `wpcalypso/no-package-relative-import` warning. 

But now that I think about it, I'm not sure if the package we import in the client code is the same we import in the tests code.

brbrr replied:
Unfortunately, I'm not sure. In the example above I can see that it uses `isEnabled` method, which does not exist in a `config` used here. 

I think that `client` import is handled by webpack setup, which is not used in e2e tests.

I'd prefer to revert this change back, but if eslint complaining about it, I guess it would be fine to left it as is.

rcanepa commented:
@brbrr I updated the PR according to your feedback.

There are 3 tests failing that

brbrr reviewed (approved):
The code looks good! 

Not sure what to do with that `config` import. Feel free to ignore my comment if you feel it does not worth reverting.

rcanepa commented:
> The code looks good!
> 
> Not sure what to do with that `config` import. Feel free to ignore my comment if you feel it does not worth reverting.

@brbrr I moved the `config` import back to the external dependencies block and I rebased the PR to solve a conflict.
